src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -386,10 +442,20 @@ def test_WINNF_FT_S_FAD_1(self, config_filename):
       # verify that retrieved cbsd dump files have correct schema
       for cbsd_record in cbsd_dump_data:
           self.assertContainsRequiredFields("CbsdData.schema.json", cbsd_record)
-          self.assertFalse("cbsdInfo" in cbsd_record)
-            
+          self.assertFalse("cbsdInfo" in cbsd_record)        
       # verify all the previous activities on CBSDs and Grants exist in the dump files
       self.assertCbsdRecord(config['registrationRequests'], grants, grant_responses, cbsd_dump_data, config['conditionalRegistrationData'])
+      # step 10 check all SAS Test Harnesses retrieve all of the data in the Full Activity Dump from the SAS UUT
+      for sas_th in config['sasTestHarnessConfigs'][1:]:
+        dump_message = self._sas.GetFullActivityDump(sas_th['serverCert'], sas_th['serverKey'])
+        # check that dump message is the same as the message retreived by the first SAS TH
+        self.assertEqual(response, dump_message)
+        # check that dump files are the same as the files retreived by the first SAS TH
+        for index, dump_file in enumerate(dump_message['files']):
+          if dump_file['recordType'] != 'coordination':                
+              downloaded_file = self._sas.DownloadFile(dump_file['url'],\
+                sas_th['serverCert'], sas_th['serverKey'])
+              self.assertEqual(hash_of_dump_file[index], hashlib.sha1(downloaded_file).hexdigest())
```

Comment:
I think it's risky to require that the SAS provide the files in exactly the same order each time. Maybe it's better to store a dictionary where the key is the URL used to retrieve the file and the value is the hash of the file (and add a note that a SAS which e.g. simply re-orders the data within the files may falsely fail this test case, just in case someone runs into trouble with that).

Reply:
yes, that's true, I updated the code, thnx.